daemons/schedulerd/pacemaker-schedulerd.c

```diff
@@ -321,19 +323,33 @@ int
 main(int argc, char **argv)
 {
     GError *error = NULL;
+    int rc = pcmk_rc_ok;
 
+    GOptionGroup *output_group = NULL;
     pcmk__common_args_t *args = pcmk__new_common_args(SUMMARY);
     gchar **processed_args = pcmk__cmdline_preproc(argv, NULL);
-    GOptionContext *context = build_arg_context(args);
+    GOptionContext *context = build_arg_context(args, &output_group);
 
     crm_log_preinit(NULL, argc, argv);
     mainloop_add_signal(SIGTERM, pengine_shutdown);
 
+    pcmk__register_formats(NULL, formats);
     if (!g_option_context_parse_strv(context, &processed_args, &error)) {
         exit_code = CRM_EX_USAGE;
         goto done;
     }
 
+    rc = pcmk__output_new(&out, args->output_ty, args->output_dest, argv);
+    if ((rc != pcmk_rc_ok) || (out == NULL)) {
+        exit_code = CRM_EX_FATAL;
+        g_set_error(&error, PCMK__EXITC_ERROR, exit_code, "Error creating output format %s: %s",
+                    args->output_ty, pcmk_rc_str(rc));
+        goto done;
+    }
+
+    pe__register_messages(out);
+    pcmk__register_lib_messages(out);
```

Comment:
Not necessary with this PR, but I don't think we need to register any messages for the main output object since it's only used with version/help. That could change when we implement the metadata output though